src/main/java/cz/cvut/kbss/study/service/repository/RepositoryUserService.java

```diff
@@ -76,6 +80,14 @@ public User getCurrentUser() {
         return securityUtils.getCurrentUser();
     }
 
+    @Transactional(readOnly = true)
+    @Override
+    public User findCurrentUser() {
+        User currentUser = securityUtils.getCurrentUser();
+        em.detach(currentUser);
```

Comment:
@blcham 
I tried implementing this method in SecurityUtils as you suggested, but the issue still occurred because, as you mentioned, we are retrieving the user within the same transaction. Only this solution resolved the problem

Reply:
Done, I removed em from the service layer and used securityUtils, but I still have some questions about why it works now..